hazelcast/src/main/java/com/hazelcast/query/impl/IndexCopyBehavior.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.query.impl;
+
+/**
+ * Defines the behavior for index copying on index read/write.
+ */
+public enum IndexCopyBehavior {
+    /**
+     * READ - Internal data structures of the index are concurrently modified without copy-on-write semantics.
+     * Index queries copy the results of a query on index read to detach the result from the source map.
+     * Should be used in index-write intensive cases. (default value).
+     */
+    READ,
```

Comment:
What about renaming these enums to something like  `COPY_ON_READ`, `COPY_ON_WRITE`, `NO_COPY`? It seemed a little more readable.

Reply:
True, I mean probably both make sense, but the name of the Enum reveals what it means, doesn't it?

Reply:
I believe `COPY_ON_READ`, `COPY_ON_WRITE` are widespread terms in this context.

Reply:
OK, fixed.